### PR TITLE
HasXxx() methods now return bool only + always require field to be defined first

### PR DESCRIPTION
**Pull request by mvorisek**
merge after: https://github.com/atk4/core/pull/233

this PR also removes option to get/set field without definition using deprecated `strict_field_check` model option

### CODE REVIEW

**Comment by romaninsh** [comment]
Check if there is anything in documentation that would need to be updated too.

**Comment by mvorisek** [comment]
> Check if there is anything in documentation that would need to be updated too.

@romaninsh I try to do that, this PR is not done yet

**Comment by romaninsh** [comment]
We might need a tag for PRs which obsolete things, so we can find them and remove obsolete features in V3.

File: src/Model.php [hunk 1]
```diff
@@ -609,37 +594,27 @@ public function addFields(array $fields, $defaults = [])
...
 
-    /**
-     * Finds a field with a corresponding name. Returns false if field not found. Similar
-     * to hasElement() but with extra checks to make sure it's certainly a field you are
-     * getting.
-     *
-     * @return Field|false
-     */
-    public function hasField(string $name)
+    public function hasField(string $name): bool
```
Comment (DarkSide666) [inline src/Model.php:604]: Why no method description at all?

File: src/Model.php [hunk 2]
```diff
@@ -609,37 +594,27 @@ public function addFields(array $fields, $defaults = [])
...
 
-    /**
-     * Same as hasField, but will throw exception if field not found.
-     * Similar to getElement().
-     *
-     * @throws \atk4\core\Exception
-     *
-     * @return Field
-     */
-    public function getField(string $name)
```
Comment (DarkSide666) [inline src/Model.php:637]: Why no method description at all?

**Review by DarkSide666 — APPROVED** [review]

File: src/Model.php (hunk 2 above)
Comment (mvorisek) [reply src/Model.php:637]: This method get's a field for you by it's name specified with string!

where is the advantage? This should be actualyl the goal ;-)

I do not say everywhere, but writing comments like these is bad. The same for describing __constructor method as "Constructor." or class X like "This is a class X for X."...

**Pull request merged**